writer.go

```diff
@@ -1498,28 +1506,27 @@ func (w *writerColumn) writePageToFilter(page Page) (err error) {
 	return err
 }
 
-func (c *writerColumn) writePageTo(size int64, writeTo func(io.Writer) (int64, error)) error {
-	buffer := c.pool.GetBuffer()
-	defer func() {
-		if buffer != nil {
-			c.pool.PutBuffer(buffer)
+func (c *writerColumn) writePageTo(size int64, writeTo func(io.Writer) (int64, error)) (err error) {
+	if c.pageBuffer == nil {
+		c.pageBuffer = c.pool.GetBuffer()
```

Comment:
I wonder if it makes sense to pass in `size` into `GetBuffer` to hint the allocator how big of a buffer it should create.

Reply:
Actually, if that was an option, it might have changed this design significantly. My assumption was that because this is in the public API we didn't want to make changes that would break code using this library.

Based on the way this buffer is being used here, we'd want to size the buffer for the whole column - I'm not sure that can be known in general. It might be better to just try to use buffer implementations that can grow gracefully.

Reply:
This was possible previously because the buffer was holding one page, and the size of the page is known. For the entire column I agree we can't really know the exact size.